controllers/warp/configReader.go

```diff
@@ -123,30 +146,58 @@ func (reader *ConfigReader) dogusReader(ctx context.Context, source config.Sourc
 	return reader.createCategories(doguCategories), nil
 }
 
-func (reader *ConfigReader) readStrings(registryKey string) ([]string, error) {
-	entry, err := reader.registry.Get(registryKey)
+func (reader *ConfigReader) readStrings(ctx context.Context, registryKey string) ([]string, error) {
+	globalConfig, err := reader.getGlobalConfig(ctx)
 	if err != nil {
-		return []string{}, fmt.Errorf("failed to read configuration entry %s from etcd: %w", registryKey, err)
+		return nil, err
+	}
+
+	entry, exists := globalConfig.Get(libconfig.Key(registryKey))
+	if !exists || entry.String() == "" {
+		return []string{}, nil
 	}
 
-	var strings []string
-	err = json.Unmarshal([]byte(entry), &strings)
+	var stringSlice []string
+	err = json.Unmarshal([]byte(entry.String()), &stringSlice)
 	if err != nil {
-		return []string{}, fmt.Errorf("failed to unmarshal etcd key to string slice: %w", err)
+		return []string{}, fmt.Errorf("failed to unmarshal global config key to string slice: %w", err)
 	}
 
-	return strings, nil
+	return stringSlice, nil
 }
 
-func (reader *ConfigReader) readBool(registryKey string) (bool, error) {
-	entry, err := reader.registry.Get(registryKey)
+func removeLegacyGlobalConfigPrefix(key string) string {
+	if strings.HasPrefix(key, "config/_global") || strings.HasPrefix(key, "/config/_global") {
+		_, after, _ := strings.Cut(key, "config/_global/")
+		return after
+	}
+
+	return key
+}
+
+func (reader *ConfigReader) getGlobalConfig(ctx context.Context) (libconfig.GlobalConfig, error) {
+	globalConfig, err := reader.globalConfigRepo.Get(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to read configuration entry %s from etcd: %w", registryKey, err)
+		return libconfig.GlobalConfig{}, fmt.Errorf("failed to get global config: %w", err)
+	}
+
+	return globalConfig, nil
+}
```

Comment:
I am always sceptical about such functions. Since the error handling has to be repeated upon calling them, what is the actual benefit of such a function? Is it just to not repeat the error message and be able to return err directly?